### ISSUE TITLE
TODOs via task list markdown doesn't support lists defined by asterisk * or plus +

### ISSUE THREAD

**Issue by diablodale** [Gruntfuggly/todo-tree#35]
github markdown lists with tasks can be defined with `-`, `*`, or `+`.
Currently, `todo-tree` only supports `-`. This is due to the default regex.

I suggest updating the default regex to support all three symbols.

Current default regex: `((//|#|<!--|;|/\\*)\\s*($TAGS)|^\\s*- \\[ \\])`

Suggested default regex: `((//|#|<!--|;|/\\*)\\s*($TAGS))|(^\\s*(-|\\*|\\+) \\[ \\])`

Unfortunately, I did not understand in your extension how your tree view removes the initial comment symbols from the comment-style todo's. Therefore, the `- []` is still in front of every markdown task. Do you see a way to remove them?

**Comment by Gruntfuggly** [comment]
Hi - the reason that the comment still get's shown is that the regex is basically made of two parts - the first section 
(**(//|#|<!--|;|/\\*)\\s*($TAGS)**|^\\s*- \\[ \\])
does the normal TODO matching, and the second part
((//|#|<!--|;|/\\*)\\s*($TAGS)|**^\\s*- \\[ \\]**)
matches markdown tasklists.

The initial comment removal relies on detecting the comment in the first section.

You could modify your configuration as follows:
```
"todo-tree.regex": "((//|#|<!--|;|/\\*|-|*|+)\\s*($TAGS))"
"todo-tree.tags": ["TODO","FIXME","\\[ \\]"]
```
Which works a bit better for your case, although it has potential to find false positives...

**Comment by diablodale** [comment]
Yes, I fully understand the regex itself. My inquiry here is twofold:
1) that the existing regex doesn't support the github task lists with + and *
2) where in your code you parse out and remove the comments so that in your TODO tree you don't see the comment symbols. Perhaps I can suggest a code change there so that similar behavior can occur with github tasks.

**Comment by Gruntfuggly** [comment]
Did you try the configuration I suggested? It works with the + and * formats.

**Comment by diablodale** [comment]
Oops, my oversight. I'll look at your approach later this week when I'm back in my office.

**Comment by Gruntfuggly** [comment]
Hi - did you manage to get this to work?
